Week3/homework/step2-6.js

```diff
@@ -5,12 +5,16 @@ const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 function flattenArray2d(arr) {
   // Replace this comment and the next line with your code
-  console.log(arr);
+  const newArr = [];
+  arr.map(element1 => element1.map(element2 => newArr.push(element2)));
+  return newArr;
 }
 
 function flattenArray3d(arr) {
   // Replace this comment and the next line with your code
-  console.log(arr);
+  const newArray = [];
+  arr.map(item1 => item1.map(item2 => item2.map(item3 => newArray.push(item3))));
```

Comment:
Logic works fine but imagine if you have to do this for a 4-D or a 5-D array. What happens in that case? Suggestion: try reading about a programming concept called recursion and see if you can apply the same here. 🙂